nemo/collections/tts/models/fastpitch.py

```diff
@@ -550,3 +538,32 @@ def input_example(self, max_batch=1, max_dim=256):
 
     def forward_for_export(self, text, pitch, pace, speaker=None):
         return self.fastpitch.infer(text=text, pitch=pitch, pace=pace, speaker=speaker)
+
+    @classmethod
+    def from_pretrained(
+        cls, 
+        model_name: str,
+        refresh_cache: bool = False,
+        override_config_path: Optional[str] = None,
+        map_location: Optional['torch.device'] = None,
+        strict: bool = True,
+        return_config: bool = False,
+        trainer: Optional['Trainer'] = None,
+        save_restore_connector: SaveRestoreConnector = None,
+    ):
+        """Custom Fastpitch's load logic to support old checkpoint."""
+        if model_name == "tts_en_fastpitch":
+            logging.warning("Please use https://github.com/NVIDIA/NeMo/tree/main/examples/tts/conf/fastpitch_align_old.yaml \
+            as config_file to avoid using deprecated AudioToCharWithDursF0Dataset class and run into errors. \
+            Recommended usage: FastPitchModel.from_pretrained(\"tts_en_fastpitch\", override_config_path=\"/path/to/fastpitch_align_old.yaml\")")
+        model = super().from_pretrained(
+            model_name,
+            refresh_cache,
+            override_config_path,
+            map_location,
+            strict,
+            return_config,
+            trainer,
+            save_restore_connector,
+            )
+        return model
```

Comment:
Ok let's remove this code then and move it to init. Do a check like `if self.ds_class_name == "AudioToCharWithPriorAndPitchDataset":` then write an appropriate error message.

Reply:
Done.